connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -185,6 +189,25 @@ public void testAdvertisedUri() {
         Assert.assertEquals("http://plaintext-localhost:4761/", server.advertisedUrl().toString());
     }
 
+    @Test
+    public void testValidateUriHost() {
+        validateUriHost(URI.create("http://localhost:8080"));
+        validateUriHost(URI.create("http://172.217.2.110:80"));
+        validateUriHost(URI.create("http://[2607:f8b0:4006:818::2004]:80"));
```

Comment:
Could we add some valid IDN hostnames here too?

Reply:
done